sdk/src/main/scala/io/horizen/forge/AbstractForger.scala

```diff
@@ -136,10 +136,36 @@ abstract class AbstractForger[
 
 
   protected def tryToCreateBlockNow(): Unit = {
+    // we are in the timer func, triggered periodically for automatic forging attempt
     val currentTime: Long = timeProvider.time() / 1000
+/* TODO remove after code review
+This could be enabled if we want to skip the forging step entirely while we are syncing, thus saving many cpu cycles. Otherwise the slot is skipped while trying to forge.
+The drawback of this approach is that we call an async func (with a future obj) for getting current tip info every timer expiration
```

Comment:
I don't think it is worthy to have every time an additional async call for a check that is useful only for a short period of the lifetime of a forger. Maybe it can be done inside tryToForgeNextBlock before calling getBranchPointInfo, so we can still save some cpu cycles without the overhead of the additional async call.

Reply:
agree with Alberto, let's remove the commented code and keep the code simple